plugins/out_opentelemetry/opentelemetry.c

```diff
@@ -723,6 +731,37 @@ static int flush_to_otel(struct opentelemetry_context *ctx,
     resource_log.n_scope_logs = 1;
     resource_logs[0] = &resource_log;
 
+    kv_size = mk_list_size(&(ctx->kv_labels));
+    attributes_list = flb_calloc(kv_size, sizeof(Opentelemetry__Proto__Common__V1__KeyValue *));
```

Comment:
The code that depends on `kv_size` holding a value higher than zero should be conditional so it doesn't run unless necessary.